### PR TITLE
Snap 3189

### PR DESCRIPTION
**Pull request by smahajan05**
## What changes were proposed in this pull request?

Added code changes to change the metric namespace

## How was this patch tested?

Manually tested

Please review http://spark.apache.org/contributing.html before opening a pull request.

Other prs

https://github.com/SnappyDataInc/snappydata/pull/1463

https://github.com/SnappyDataInc/snappy-store/pull/524

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala [hunk 1]
```diff
@@ -133,7 +133,11 @@ private[spark] class MetricsSystem private (
 
     if (instance == "driver" || instance == "executor") {
       if (metricsNamespace.isDefined && executorId.isDefined) {
-        MetricRegistry.name(metricsNamespace.get, executorId.get, source.sourceName)
+        if (source.sourceName.contains("TIBCO ComputeDB")) {
```
Comment (snappy-sachin) [inline core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala]: What name it will use if product is OSS (i.e SnappyData)?

**Review by snappy-sachin — APPROVED** [review]
Looks good, please check one concern mentioned in comment.

File: core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala (hunk 1 above)
Comment (smahajan05) [reply core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala]: Added code change to provide namespace based on product i.e enterprise or OSS

File: core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala [hunk 2]
```diff
@@ -133,7 +133,12 @@ private[spark] class MetricsSystem private (
 
     if (instance == "driver" || instance == "executor") {
       if (metricsNamespace.isDefined && executorId.isDefined) {
-        MetricRegistry.name(metricsNamespace.get, executorId.get, source.sourceName)
+        if (source.sourceName.contains("TIBCO ComputeDB") ||
+            source.sourceName.contains("SnappyData")) {
+          MetricRegistry.name("", "", source.sourceName)
+        } else {
+          MetricRegistry.name(metricsNamespace.get, executorId.get, source.sourceName)
```
Comment (kneeraj) [inline core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala:144]: Why else is required. If for Spark side metrics then please add a comment. Also please put a comment as to why you are ignoring the metricsNamespace and executorId.

**Review by kneeraj — APPROVED** [review]
Please add dunit tests before merge.

File: core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala (hunk 2 above)
Comment (smahajan05) [reply core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala:144]: Added suggested changes

**Pull request merged**